Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/approximate_convex_decomposition.h

```diff
@@ -0,0 +1,1639 @@
+        pts.reserve(pts.size() + cj.points.size());
+        std::copy(cj.points.begin(), cj.points.end(), std::back_inserter(pts));
+        convex_hull_3(pts.begin(), pts.end(), ch.points, ch.indices);
+
+        ch.volume = volume<GeomTraits>(ch.points, ch.indices);
+
+        ch.volume_error = m.volume_error = CGAL::abs(ci.volume + cj.volume - ch.volume) / hull_volume;
+        queue.push(std::move(m));
+      }
+      else {
+        Merged_candidate m(i, j);
+        Bbox_3 bbox = box_union(ci.bbox, cj.bbox).bbox();
+        m.ch = -1;
+        m.volume_error = CGAL::abs(ci.volume + cj.volume - bbox.x_span() * bbox.y_span() * bbox.z_span()) / hull_volume;
+        queue.push(std::move(m));
+      }
+    }
+  }
+
+  while (!queue.empty() && keep.size() > max_convex_hulls) {
+    Merged_candidate m = queue.top();
+    queue.pop();
+
+    auto ch_a = hulls.find(m.ch_a);
+    if (ch_a == hulls.end())
+      continue;
+
+    auto ch_b = hulls.find(m.ch_b);
+    if (ch_b == hulls.end())
+      continue;
+
+    if (m.ch == -1)
+      do_merge(m);
+
+    keep.erase(m.ch_a);
+    keep.erase(m.ch_b);
+
+    hulls.erase(ch_a);
+    hulls.erase(ch_b);
+
+    const Convex_hull_candidate<GeomTraits>& cj = hulls[m.ch];
+
+    for (std::size_t id : keep) {
+      const Convex_hull_candidate<GeomTraits>& ci = hulls[id];
+      if (CGAL::do_intersect(ci.bbox, cj.bbox)) {
+        Merged_candidate merged(id, m.ch);
+
+        merged.ch = num_hulls++;
+        Convex_hull_candidate<GeomTraits>& ch = hulls[merged.ch];
+        ch.bbox = box_union(ci.bbox, cj.bbox);
+        std::vector<Point_3> pts(ci.points.begin(), ci.points.end());
+        pts.reserve(pts.size() + cj.points.size());
+        std::copy(cj.points.begin(), cj.points.end(), std::back_inserter(pts));
+        convex_hull_3(pts.begin(), pts.end(), ch.points, ch.indices);
+
+        ch.volume = volume<GeomTraits>(ch.points, ch.indices);
+
+        ch.volume_error = merged.volume_error = CGAL::abs(ci.volume + cj.volume - ch.volume) / hull_volume;
+        queue.push(std::move(merged));
+      }
+      else {
+        Merged_candidate merged(id, m.ch);
+        Bbox_3 bbox = box_union(ci.bbox, cj.bbox).bbox();
+        merged.ch = -1;
+        merged.volume_error = CGAL::abs(ci.volume + cj.volume - bbox.x_span() * bbox.y_span() * bbox.z_span()) / hull_volume;
+        queue.push(std::move(merged));
+      }
+    }
+
+    keep.insert(m.ch);
+  }
+
+  num_hulls = 0;
+
+  for (std::size_t i : keep)
+    candidates.push_back(std::move(hulls[i]));
+}
+
+} // namespace internal
+/**
+ * \ingroup PMP_convex_decomposition_grp
+ *
+ * \brief approximates the input mesh by a number of convex hulls. The input mesh is voxelized and the voxels intersecting with the mesh are labeled as surface.
+ *        The remaining voxels are labeled as outside or inside by floodfill, in case the input mesh is closed, or by axis-aligned ray shooting, i.e., along x,
```

Comment:
```suggestion
 *        The remaining voxels are labeled as outside or inside by flood fill, in case the input mesh is closed, or by axis-aligned ray shooting, i.e., along x,
```